controllers/autohealconfig_controller.go

```diff
@@ -47,16 +67,397 @@ type AutohealConfigReconciler struct {
+		// httpclient := http.Client{}
+		repoUrl, err := url.ParseRequestURI(fmt.Sprintf("%s/contents/%s", repo.RepositoryURL, endpoints.Filename))
+		if err != nil {
+			return nil, err
+		}
+
+		req, err := http.NewRequest(http.MethodGet, repoUrl.String(), nil)
+		if err != nil {
+			log.Errorf("Unable to create new http request")
+			return nil, err
+
+		}
+		if repo.Tag != "" {
+			q := req.URL.Query()
+			q.Add("ref", repo.Tag)
+			req.URL.RawQuery = q.Encode()
+		}
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", repo.AccessToken))
+		req.Header.Set("Accept", "application/vnd.github.v3.raw")
+
+		resp, err := r.httpClient.Do(req)
+		if err != nil {
+			log.Errorf("Unable to send http request")
+			return nil, err
+
+		}
+		if resp.StatusCode != http.StatusOK {
+			log.Infof("Non 200 response back, got status code %d ", resp.StatusCode)
+			return nil, nil
+		}
+
+		bytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Errorf("Unable to read response body %s", err)
+			return nil, err
+
+		}
+		err = resp.Body.Close()
+		if err != nil {
+			return nil, err
+		}
+
+		err = yaml.Unmarshal(bytes, &scenario)
+		if err != nil {
+			log.Errorf("Unable to parse yaml scenarios from config repo %s", err)
+			return nil, err
+
+		}
+		scenarios = append(scenarios, scenario)
+
+	}
+	return scenarios, nil
+}
+
+// createConfigMaps Creates the config maps based on the scenarios gotten from github
+func (r *AutohealConfigReconciler) createConfigMaps(autoheal *v1alpha1.AutohealConfig, scenario []v1alpha1.AutohealConfigScenario, ctx context.Context) error {
+	log.Info("Stage: Creating/updating configmaps")
+
+	for _, scenarios := range scenario {
+		configmap := core.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      scenarios.Name,
+				Namespace: autoheal.Namespace,
+			},
+		}
+
+		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, &configmap, func() error {
+			if configmap.Data == nil {
+				configmap.Data = make(map[string]string)
+			}
+			configmap.Labels = map[string]string{
+				"app": "autoheal",
+			}
+			configmap.Data["query"] = scenarios.Query
+			configmap.Data["label"] = scenarios.Label
+			configmap.Data["script"] = scenarios.Script
+			configmap.Data["gracePeriod"] = scenarios.GracePeriod
+			return nil
+
+		})
+		if err != nil {
+			log.Errorf("Unable to create configmap")
+			return err
+
+		}
+	}
+	return nil
+}
+
+// updateStatus Updates the controller to how long until next requeue of the reconciller depening on if its a error or not
+func (r *AutohealConfigReconciler) updateStatus(autoheal *v1alpha1.AutohealConfig) (ctrl.Result, error) {
+	err := r.Client.Status().Update(context.Background(), autoheal)
+	if err != nil {
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: RequeueDelayError,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: RequeueDelaySuccess,
+	}, nil
+}
+
+// deleteConfigSecret Delests the config secret with creds to sync to the repo
```

Comment:
```suggestion
// deleteConfigSecret deletes the config secret containing the creds to sync to the repo
```

Reply:
Actually just on this function: do we want to delete the config secret? For dev purposes, it can be a bit of a pain because we have to re-create it every time. Maybe for real-world use, we might want to delete it. I'm not really sure.

Reply:
Great question, let me do some testing to see if removing causes un wanted finalizes to stick around on the secret

Reply:
i can remove it and we can always add it back down the line if we see a need for it